lib/hooks.ts

```diff
@@ -30,13 +30,17 @@ export function useProducts() {
     };
 }
 
-export function useProductList() {
+export function useProductList(page?: number, limit?: number) {
     const encodedContext = useSession()?.context;
+    const pageQuery = page ? `&page=${page}` : '';
+    const limitQuery = limit ? `&limit=${limit}` : '';
+    const query = `${pageQuery}${limitQuery}`;
     // Use an array to send multiple arguments to fetcher
-    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', encodedContext] : null, fetcher);
+    const { data, error, mutate: mutateList } = useSWR(encodedContext ? ['/api/products/list', encodedContext, query] : null, fetcher);
 
     return {
-        list: data,
+        list: data ? data.data : [],
```

Comment:
same here, let's simplify:
```suggestion
        list: data.data,
```

Reply:
@bc-chaz I think this or something similar may actually be necessary now because of how the shape of the data has changed. Since we have two separate items now inside of the data object and need to reference them individually, we run into an issue with SWR where `data` is undefined until it is loaded. `list: data.data,` results in a runtime error `TypeError: Cannot read properties of undefined (reading 'data')`. This is what I was trying to work around with the optional chaining before and now this. 

Before, when `data` only contained the product list array, we weren't referencing anything inside of it (`list: data` as you can see in the diff) so it didn't matter that `data` is undefined during the loading state.

Reply:
err so currently, `data` is a destructure of the response - https://github.com/bigcommerce/sample-app-nodejs/blob/main/pages/api/products/list.ts#L13. SWR might be doing something internal to account for this. If we're getting an error here, I'd recommend adding in optional chaining and allowing an undefined state:
```jsx
return {
    list: data?.data,
    meta: data?. meta,
}
```